snippets/javascript.json

```
@@ -653,5 +653,598 @@
+        "\tbandIds: [${2:3}, ${3:2}, ${4:1}]",
+        "});"
+      ],
+      "description": "Create an ImageryTileLayer that references a Cloud Optimized GeoTiff(COG) file. A Cloud Optimized GeoTiff (COG) file is a regular GeoTiff file that is hosted on CORS enabled HTTPS server. AMD path: esri/layers/ImageryLayer | ESM path: @arcgis/core/layers/ImageryLayer.js",
+      "prefix": "ImageryTileLayerwithCOG"
+    },
+  "Create a Transposed Multidimensional Imagery Tile Layer": {
+      "body": [
+        "new MultidimensionalImageryTileLayer({",
+        "\turl: \"${1:https://tiledimageservices.arcgis.com/V6ZHFr6zdgNZuVG0/arcgis/rest/services/NLDAS2011_daily_wind_magdir/ImageServer}\",",
+        "\ttitle: \"${2:2011 wind - 10 meters above surface}\",",
+        "\trenderer: {",
+        "\t\ttype: \"vector-field\",",
+        "\t\tstyle: \"${3:beaufort-m}\",",
+        "\t\tflowRepresentation: \"${4:flow-to}\",",
+        "\t\tsymbolTileSize: ${5:30},",
+        "\t\tvisualVariables: [",
+        "\t\t\t{",
+        "\t\t\t\ttype: \"size\",",
+        "\t\t\t\tfield: \"${6:Magnitude}\",",
+        "\t\t\t\tmaxDataValue: ${7:32},",
+        "\t\t\t\tmaxSize: \"${8:60px}\",",
+        "\t\t\t\tminDataValue: ${9:0.04},",
+        "\t\t\t\tminSize: \"${10:20px}\"",
+        "\t\t\t},",
+        "\t\t\t{",
+        "\t\t\t\ttype: \"rotation\",",
+        "\t\t\t\tfield: \"${11:Direction}\",",
+        "\t\t\t\trotationType: \"${12:geographic}\"",
+        "\t\t\t}",
+        "\t\t]",
+        "\t}",
+        "});"
+      ],
+      "description": "Create a Transposed Multidimensional Imagery Tile Layer. Transposed Multidimensional Imagery Tile Layers are created for optimizing performance when accessing pixel values across all multidimensional slices rather than for visualization at different map scales. AMD path: esri/layers/ImageryLayer | ESM path: @arcgis/core/layers/ImageryLayer.js",
+      "prefix": "TransposedMultidimensionalImageryTileLayer"
+    },
+  "Add WebTileLayer": {
+      "body": [
+        "new WebTileLayer({",
+        "\turlTemplate: \"${1:https://{subDomain}.tile.opentopomap.org/{level}/{col}/{row}.png}\",",
+        "\tsubDomains: [\"${2:a}\", \"${3:b}\", \"${4:c}\"],",
+        "\tcopyright: \"${5:https://www.openstreetmap.org/copyright}\" ",
+        "});"
+      ],
+      "description": "Create a WebTileLayer with a service URL. WebTileLayer provides a simple way to add non-ArcGIS Server map tiles as a layer to a map. The constructor takes a URL template that usually follows a pattern of http://some.domain.com/{level}/{col}/{row}/ where level corresponds to a zoom level, and column and row represent tile column and row, respectively. AMD path: esri/layers/WebTileLayer | ESM path: @arcgis/core/layers/WebTileLayer.js",
+      "prefix": "WebTileLayer"
+    },
+  "Add WMS Layer": {
+    "body": [
+    "new WMSLayer({",
+    "\turl: \"${1:service url}\"",
+    "});"
```

```suggestion
    "});",
    "await wmsLayer.load();"
```
If I remember correctly, in this case, calling load() is important.
Understanding the loadable [pattern](https://developers.arcgis.com/javascript/latest/loadable/)